hazelcast/src/main/java/com/hazelcast/spi/discovery/DiscoveryStrategy.java

```diff
@@ -91,5 +91,5 @@
      * @return a map of discovered metadata as provided by the runtime environment
```

Comment:
Not related to this PR but the javadoc seems wrong:
```
* Information discovered from this method are shaded into the {@link com.hazelcast.core.Member}s
     * attributes. Existing attributes will not be overridden, that way local attribute configuration
     * overrides provided metadata.
```
I see the metadata overriding the attributes. Can you confirm and change the javadoc while you're at it?

Reply:
Yes, looks like they are overridden. I updated the Javadoc. Also, added a test in `DiscoverySpiTest.java` to verify member attributes are overridden by discovery attributes.